jsonrpc/eth_endpoint.go

```diff
@@ -1018,3 +1133,38 @@ func (e *Eth) FeeHistory(blockCount argUint64, newestBlock BlockNumber,
 
 	return result, nil
 }
+
+// Sign calculates an ECDSA signature for:
+// keccak256("\x19Ethereum Signed Message:\n" + len(message) + message).
+//
+// Note, the produced signature conforms to the secp256k1 curve R, S and V values,
+// where the V value will be 27 or 28 for legacy reasons.
+//
+// The account associated with addr must be unlocked.
+//
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign
+func (e *Eth) Sign(_ types.Address, buf argBytes) (interface{}, error) {
+	tx := &types.Transaction{}
+	if err := tx.UnmarshalRLP(buf); err != nil {
+		return nil, err
+	}
+
+	if e.ecdsaKey == nil {
+		return nil, errMissingPrivateKey
+	}
+
+	cryptoECDSAPrivKey, err := polyWallet.AdaptECDSAPrivKey(e.ecdsaKey)
+	if err != nil {
+		return nil, err
+	}
+
+	signedTx, err := e.txSigner.SignTx(tx, cryptoECDSAPrivKey)
+	if err != nil {
+		return nil, err
+	}
+
+	data := signedTx.MarshalRLP()
+	data[64] += 27 // Transform V from 0/1 to 27/28 according to the yellow paper
```

Comment:
I think this is not correct. It's correct when we have a byte array of the signature, where the 65th byte is V. However, when marshalling the entire transaction, I'm not sure that the 65th byte is associated with V.
@Stefan-Ethernal correct me if i am wrong.

Reply:
Looks like you're right